OHHTTPStubs/UnitTests/Test Suites/SwiftHelpersTests.swift

```diff
@@ -454,6 +454,72 @@ class SwiftHelpersTests : XCTestCase {
 #endif
   }
 
+#if swift(>=3.0)
+  func testHasJsonBodyIsTrue() {
+    let jsonObjects = [
+      // Exact match
+      (["foo": "bar", "baz": 42, "qux": true],
+       ["foo": "bar", "baz": 42, "qux": true]),
+      // Changed attribute order
+      (["foo": "bar", "baz": 42, "qux": true],
+       ["qux": true, "foo": "bar", "baz": 42]),
+      // Nested objects
+      (["foo": "bar", "baz": ["qux": true, "quux": ["spam", "ham", "eggs"]]],
+       ["foo": "bar", "baz": ["qux": true, "quux": ["spam", "ham", "eggs"]]]),
+      // Nested objects with changed attribute order
+      (["foo": "bar", "baz": ["qux": true, "quux": ["spam", "ham", "eggs"]]],
+       ["foo": "bar", "baz": ["quux": ["spam", "ham", "eggs"], "qux": true]]),
+    ]
```

Comment:
The problem with that approach is that dictionaries have no guaranteed order (and that the whole point of your PR). So there's no guarantee that Swift won't make all those dictionaries exactly the same internally, and then making the JSON serialized version of all those exactly the same (e.g. one might even imagine that `JSONSerialization.data(withJSONObject:)` could iterate over the dictionaries' keys in alphabetical order…)

To solve that, I think you'll instead need to use the already-converted-to-JSON String representations as a test set instead.

Which would have the additional benefit of proving that your matcher is also robust against different indentations of JSON bodies.

One way to do that is to use string literals of JSON representations to test
```
let jsonObjects = [
  "{ \"foo\": \"bar\", \n\"baz\": 42,      \"qux\": true }",
…
```
But for readability of the unit tests, maybe outsourcing those texts/JSON representation in fixture files would be nicer.

Reply:
Your suggestion makes sense, but just to be clear; what you're suggesting is the following:

```
    let jsonStrings = [
      ("{ \"foo\": \"bar\", \"baz\": 42, \"qux\": true }",
       ["foo": "bar", "baz": 42, "qux": true])
      // Etc...
    ]
    
    for (jsonString, expectedJsonObject) in jsonStrings {
      var req = URLRequest(url: URL(string: "foo://bar")!)
      req.httpBody = jsonString.data(using: .utf8)
      let matchesJsonBody = hasJsonBody(expectedJsonObject)(req)
      
      XCTAssertTrue(matchesJsonBody)
    }
```

Reply:
Yup, exactly 👍